### PR TITLE
People: Add email followers filter to people page

### PR DESCRIPTION
**Pull request by leandroalonso**
Original PR: #17830

See #6969

Adds an email filter to the people page.

**Related PRs:** 

- wordpress-mobile/WordPressKit-iOS#477
- wordpress-mobile/WordPressKit-iOS#480

### To test:

- Login to an account with email followers (or invite one)
- On the site tab, scroll down to Configure and tap on People
- Tap on "Email Followers" to see your site's email followers
- Tap on a email follower and view the email follower's page
- Remove the follower if you would like to test the removal

## Screenshots
|||
|-|-|
|  |  |
|  |  |

## Regression Notes
1. Potential unintended areas of impact
Potentially the other filters

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Verified those filters were still working correctly after adding the email filter.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17854-b7c15a2</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/b40237b4-59df-48e6-9218-834ba00b529d/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=TsQ6%2B4rLLgaxOlKcuO8nbeMdxkmmvfvW2Ca37Iawntk%3D&se=2022-02-03T14%3A07%3A57Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17854-b7c15a2</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/78d8862a-5ee4-43d1-ab0e-f8616bc2bdb2/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=SJI8tq3d%2F2sH6zqi8%2FoYm5Fto2m%2Fq0TvoExspmbLkMQ%3D&se=2022-02-03T14%3A09%3A55Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

File: WordPress/Classes/Services/PeopleService.swift [hunk 1]
```diff
@@ -182,6 +204,30 @@ struct PeopleService {
...
+    ///
+    func deleteEmailFollower(_ person: EmailFollower, success: (() -> Void)? = nil, failure: ((Error) -> Void)? = nil) {
+        guard let managedPerson = managedPersonFromPerson(person) else {
+            return
+        }
+
+        remote.deleteEmailFollower(siteID, userID: person.ID, success: {
+            success?()
+        }, failure: { error in
+            DDLogError("### Error while deleting follower \(person.ID) from blog \(self.siteID): \(error)")
```
Comment (momo-ozawa) [inline WordPress/Classes/Services/PeopleService.swift]: [nit]
```suggestion
            DDLogError("Error while deleting follower \(person.ID) from blog \(self.siteID): \(error)")
```

File: WordPress/Classes/ViewRelated/People/PeopleViewController.swift [hunk 2]
```diff
@@ -70,6 +70,8 @@ class PeopleViewController: UITableViewController, UIViewControllerRestoration {
         //
         switch filter {
         case .followers:
+            fallthrough
+        case .email:
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/People/PeopleViewController.swift]: [nit]

imo `case .followers, .email:` seems more readable

**Review by momo-ozawa — APPROVED** [review]
Works as described.

File: WordPress/Classes/ViewRelated/People/PeopleViewController.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/People/PeopleViewController.swift]: @wargcm wdyt? Double-checking just in case.

Comment (wargcm) [reply WordPress/Classes/ViewRelated/People/PeopleViewController.swift]: It's a good call, it was something I overlooked. `fallthrough` probably makes more sense if you're combining some of the cases whereas this is just two cases with the same result.

Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/People/PeopleViewController.swift]: Done in b7c15a2e2ee31015673227c26c277a302fe0cd20

**Pull request merged**